Mathlib/Topology/Algebra/Valued/ValuativeRel.lean

```diff
@@ -120,21 +156,67 @@ instance (priority := low) isTopologicalAddGroup : IsTopologicalAddGroup R := by
   · simpa [ContinuousAt] using (cts_add.1 x₀).continuousAt (x := (0 : R))
   · simpa [ContinuousAt] using (cts_add.1 (-x₀)).continuousAt (x := x₀)
 
-instance (priority := low) : IsTopologicalRing R :=
-  letI := IsTopologicalAddGroup.toUniformSpace R
-  letI := isUniformAddGroup_of_addCommGroup (G := R)
-  inferInstance
+instance (priority := low) : IsTopologicalRing R := by
+  convert (valuation R).subgroups_basis.toRingFilterBasis.isTopologicalRing
+  rw [TopologicalSpace.ext_iff_nhds]
+  intro x
+  ext s
+  simp [(RingSubgroupsBasis.hasBasis_nhds _ _).mem_iff, mem_nhds_iff, Valuation.ltAddSubgroup,
+    neg_add_eq_sub]
+
+/-- A ring with a topological additive structure and a valuative relationship is
+a uniform space made up of entourages of the form `{ (x, y) | v (y - x) < γ }`.
+However, this is not a global instance to prevent timeouts in typeclass inference,
+since otherwise, TC search for `UniformSpace R` will start exploring `IsValuativeTopology R`. -/
+local instance : UniformSpace R := IsTopologicalAddGroup.toUniformSpace R
```

Comment:
This is really weird. Shouldn't you just assume `[UniformSpace R] [IsUniformAddGroup R]`?

Reply:
We don't have API to show that when `IsTopologicalGroup`, then any UniformSpace is necessarily equal to the one from `IsTopologicalGroup.toUniformSpace`.

In fact, in the `IsUniformGroup` file, there is also a
```
attribute [local instance] IsTopologicalGroup.toUniformSpace
```

Reply:
Because you need to assume `IsUniformAddGroup` and not `IsTopologicalGroup`; the latter is not enough. The lemma you want is `uniformity_eq_comap_nhds_zero`.

Reply:
Because that is used to show that `IsTopologicalAddGroup.toUniformSpace` gives the correct uniformity?

But downstream lemma should work for any uniformity that is compatible.